doc/cephfs/multimds.rst

```diff
@@ -311,3 +311,114 @@ the ``bal_rank_mask`` should be set to ``0x0``. For example:
+due to insufficient MDS resources like MDS cache memory. Therefore,
+the ceph.dir.bal.mask can dynamically balance the workload of this large
+directory within a specific subset of ranks.
+
+
+The ``ceph.dir.bal.mask`` can also be a useful option for fine-tunning performance,
+particularly in scenarios involving large directories like ``/home/images``
+and ``/home/backups`` within the file system.
+While the mdsmap’s ``bal_rank_mask`` isolates the entire ``/`` directory to
+specific ranks, it can affect performance due to each other's migration overhead.
+For example, if the mdsmap’s ``bal_rank_mask`` is set to ``0xf`` and large directories
+like ``/home/images`` and ``/home/backups`` exist, the load on ``/home/images``
+instantaneously increases and metadata distribution occurs across ranks 0 to 3.
+Thus, users of ``/home/backups`` may be affected by noisy neighbors unnecessarily.
+By distributing the two directories to MDS rank 0,1 group and MDS rank 2,3 group
+respectively through the ``ceph.dir.bal.mask``, metadata service can be provided
+without affecting the performance of other concurrent workloads.
+
+
+This option can be set via:
+
+::
+
+    setfattr -n ceph.dir.bal.mask -v 1,2 /home/images
+    setfattr -n ceph.dir.bal.mask -v 3,4 /home/backups
+
+
+``/homes/images`` and ``/home/backups`` are distributed within mds rank 1~2 and 3~4, respectively.
+
+Similar to ``ceph.dir.pin``, the ``ceph.dir.bal.mask`` is also inherited from its closest parent.
+This manner involves configuring the ``ceph.dir.bal.mask`` for a directory, which consequently
+impacts all its subordinate items. However, the parent ``ceph.dir.bal.mask`` can be overriden
+by setting the child directory’s value. For example:
+
+::
+
+    mkdir -p /home/images/jpg
+    setfattr -n ceph.dir.bal.mask -v 1,2 /home/images
+    # /home/images and /home/images/jpg are now freely distributed within mds rank 1 and 2
+    setfattr -n ceph.dir.bal.mask -v 3,4 /home/images/jpg
+    # /home/images/jpg are distributed within mds rand 3 and 4
+
+
+The option can be unset via:
+
+::
+
+    setfattr -n ceph.dir.bal.mask -v -1 /home/images/jpg
+
+The value of the ``ceph.dir.bal.mask`` for  ``/home/images/jpg`` is unset and replaced with their inherited value from
+its nearest parent. If the ``ceph.dir.bal.mask`` for /home/images is configured with
+a valid value, ``/home/images/jpg`` will be distributed according to its parent’s settings.
+
+This can override the mdsmap’s ``bal_rank_mask``. For example:
+
+::
+
+    cephfs fs set cephfs bal_rank_mask 0xf
+
+
+Initially, the balancer dynamically partitions the file system within MDS rank
+0 to 3 because the ``bal_rank_mask`` is set to ``0xf``.
+
+If the ``ceph.dir.bal.mask`` for the root ``/`` directory is set to ``0,1,2,3,4,5``,
+the ``bal_rank_mask`` will be overridden. In this way, the balancer dynamically
+distributes the unpinned subtrees of the root ``/`` directory from ``0`` to ``5`` mds ranks.
```

Comment:
```bal_rank_mask``` does not need to be deprecated. If detailed tuning is required according to the user's preference, ```ceph.dir.bal.mask``` xattr can be used. Otherwise, ```bal_rank_mask``` can be used for simplicity.

Reply:
Perhaps add a note about that, literally:

```
.. note:: Users may prefer to use the file system setting `bal_rank_mask` ...
```

Reply:
Added. https://github.com/ceph/ceph/compare/9b04cd83fb615464888de665b9933d37724d4a28..ac1ab14f09a789361f9a5c72e448ae85f5b55fa8#diff-cc9cf0d49df0c5644ecb9c98d1f7ac1d307b05837bfdabc6761c65691c6f8dc1R407